src/boot/efi/pe.c

```diff
@@ -224,6 +253,57 @@ static void pe_locate_sections(
                 }
 }
 
+static bool pe_use_this_dtb(
+                const PeSectionHeader section_table[],
+                size_t n_section_table,
+                void *base,
+                size_t dtb_offset,
+                size_t dtb_size,
+                size_t section_nb) {
+        EFI_STATUS err;
+
+        static const void *cached_base = NULL;
+        static const Device *cached_device = NULL;
+
+        const uint8_t *dtb = (const uint8_t *) base + dtb_offset;
+
+        err = devicetree_match(dtb, dtb_size);
+
+        /* Firmware does not provide the devicetree, so try matching against a list from .hwids section */
+        if (err == EFI_UNSUPPORTED) {
+                static const char *const hwids_section_name[] = { ".hwids", NULL };
+                PeSectionVector hwids_section = {};
+                pe_locate_sections(
+                                section_table,
+                                n_section_table,
+                                hwids_section_name,
+                                PTR_TO_SIZE(base),
+                                &hwids_section);
+                if (hwids_section.memory_size == 0)
+                        log_error_status(err, "HWIDs section is missing, no DT blob will be selected");
```

Comment:
just do `return false;` here?

exit earl is good, and means you can drop the indentation below